flink-streaming-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraTupleWriteAheadSink.java

```diff
@@ -132,11 +132,15 @@ protected void sendValues(Iterable<IN> values, long timestamp) throws Exception
 		}
 		try {
 			while (updatesSent != updatesConfirmed.get()) {
+				if (exception.get() != null) { // verify that no query failed until now
+					LOG.warn("Sending a value failed.", exception.get());
+					break;
+				}
 				Thread.sleep(100);
 			}
```

Comment:
Could we try to replace the busy waiting loop with a wait-notify construct? That would decrease latency and wouldn't burn unnecessary CPU cycles. We could sync on the `updateConfirmed` variable.